elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,283 @@
+*** severity
+*** server address
+*** port
+*** setting TCP
+*** setting SSL
+*** setting socket factory
+*** setting hostname
+** The current attributes available in the Elytron schema, which will be used for RFC 5424, are:
+*** server-address
+*** port
+*** transport type
+*** message format
+*** hostname (?)
+*** ssl-context
+** Additional attributes which will need to be added are:
+*** include timeQuality
+*** include origin
+*** include meta
+*** facility values
+*** priority mapping from facility and severity values, in accordance with https://tools.ietf.org/html/rfc5424#page-9[RFC 5424 Section 6.2.1]
+** The following attributes will also be available, but should be discarded if timeQuality is not specified:
+*** tzKnown
+*** isSynced
+*** syncAccuracy
+** The following attributes will also be available, but should be discarded if origin is not specified:
+*** ip
+*** enterpriseId
+*** software
+*** swVersion
+** The following attributes will also be available, but should be discarded if meta is not specified:
+*** sequenceId
+*** sysUpTime
+*** language
+
+* Support for RFC 3164 should be added, based off of the implementation for RFC 5424 above, as such should also be added to the ```syslogAuditLogType``` in the Elytron schema
+** The ```syslog-format``` attribute described in the RFC 5424 section above should also support a value of ```RFC3164```
+** All parts of RFC 3164 are included in the RFC 5424 specification, as such the support for RFC 3164 should be only
+changing the message format and not require any additional implementation
+** The implementation of the RFC 3164 message format can be implemented based on the https://github.com/kabir/tamper-detecting-audit-log/blob/master/src/main/java/org/jboss/audit/log/TcpSyslogHandler.java[tamper-detecting-audit-log]
+
+* Support for user formatted strings should be added
+** Currently, this feature does not have a way to be implemented
+
+* There should be the ability to specify a Java class should be logged by using an @Audit Java annotation, similarly to what was available in Picketbox
+
+* There should be the ability to configure routing, and create a custom filter allowing only a selection of events to be logged, based on several criteria, listed in the below table
+** In each case, the user should be able to specify the criteria, the value(s), and a list of configuration changes from the default values, such as a different destination
+** Specifying the criteria, the value(s), and the configuration changes should be done as follows:
+*** When the programmatic approach is used, a method call should be available to the endpoint which will accept a given criteria, the corresponding values, and the filtering options
+*** When the markup approach is used, the inputs should be specified via attributes ```criteria="supported_criteria"```, ```values="respective_value_input"```, ```filtering="option1=change,option2=change,..."```.
```

Comment:
Where will be the markup approach used? How can user use it?

Reply:
By markup, I refer to using the XML files, i.e. standalone.xml. This would be similiar to the management audit logging described in https://docs.jboss.org/author/display/WFLY8/Audit+logging and also similar to the following code snippet:
`<audit-logging><file-audit-log format="JSON" name="local-audit" path="audit.log" relative-to="jboss.server.log.dir"/></audit-logging>`

Reply:
Thank you for explanation. However, editing of XML files no longer recommended way how to configure the server. Could you please instead of XML reference the model changes?

Reply:
Ah okay, I didn't realize this was no longer recommended (this is my first time modifying the model in detail). I'll change this section to reference the model changes